### PR TITLE
Add functionality to purge errored tasks

### PR DESCRIPTION
**Pull request by AlecRosenbaum**
Related Issue: https://github.com/closeio/tasktiger/issues/137

This change adds a `TaskTiger.purge_errored_tasks` method that makes it much easier to (surprise, surprise) purge errored tasks. Arguments allow the caller to both specify specific queues, and specify limits as to not overload redis in one method call.

Example method call:
```python
limit = 10000
n_processed = limit
while n_processed == limit:
    n_processed = tiger.purge_errored_tasks(queues=['default'], limit=limit)
    time.sleep(.5)
```

### CODE REVIEW

File: tasktiger/__init__.py [hunk 1]
```diff
@@ -482,6 +485,108 @@ def get_queue_stats(self):
...
+
+        def errored_tasks():
+            queues_with_errors = self.connection.smembers(self._key(ERROR))
+            for queue in queues_with_errors:
+                if not match(queue):
+                    continue
+
+                skip = 0
+                total = None
+                task_limit = limit or 1000
```
Comment (jkemp101) [inline tasktiger/__init__.py]: This seems a little weird setting a limit of `0` would end up being `1000`. Maybe just assert `limit > 0`?

Comment (AlecRosenbaum) [reply tasktiger/__init__.py]: I've added an additional `assert` with the other argument validation

File: tasktiger/tasktiger.py [hunk 2]
```diff
@@ -466,6 +475,92 @@ def get_queue_stats(self):
...
+                    for task in tasks:
+                        if (
+                            last_execution_before
+                            and task.ts
+                            and task.ts > last_execution_before
+                        ):
+                            continue
+                        yield task
+                    skip += task_limit
+                    total = total_tasks
```
Comment (jkemp101) [inline tasktiger/tasktiger.py]: I haven't tested this to confirm but it looks like the overall limit is being applied to the number of tasks we look at versus the number of tasks we delete. For instance, if we had 1001 total errored tasks and the first 1000's `task.ts` was greater than `last_execution_before` but the 1001 task wasn't, I don't think we would delete the 1001 task if `purge_errored_tasks()` was called with `limit=1000`.

File: README.rst [hunk 3]
```diff
@@ -720,6 +720,23 @@ reported to Rollbar. Here is a custom worker launch script:
...
+this up as a periodic task as follows:
+
+..code:: python
+  from tasktiger import TaskTiger, periodic
+
+  tiger = TaskTiger()
+
+  @tiger.task(schedule=periodic(hours=1))
+  def purge_errored_tasks():
+      tiger.purge_errored_tasks(limit=1000)
```
Comment (jkemp101) [inline README.rst]: I think I would add `last_execution_before` to the example within something like older than 3 months.

**Pull request merged**